pkg/loop/internal/relayer/pluginprovider/contractreader/contract_reader.go

```diff
@@ -121,6 +134,27 @@ func DecodeVersionedBytes(res any, vData *pb.VersionedBytes) error {
 			return fmt.Errorf("%w: %w", types.ErrInternal, err)
 		}
 		err = dec.Unmarshal(vData.Data, res)
+	case ValuesEncodingVersion:
+		protoValue := &valuespb.Value{}
+		err = proto.Unmarshal(vData.Data, protoValue)
+		if err != nil {
+			return fmt.Errorf("%w: %w", types.ErrInvalidType, err)
+		}
+
+		value, err := values.FromProto(protoValue)
+		if err != nil {
+			return fmt.Errorf("%w: %w", types.ErrInvalidType, err)
+		}
+
+		valuePtr, ok := res.(*values.Value)
+		if ok {
+			*valuePtr = value
+		} else {
+			res, err = value.Unwrap()
```

Comment:
this will only assign the local variable res.  You would do

value.UnwrapTo(res)